src/main/java/seedu/address/commons/util/DateTimeUtil.java

```diff
@@ -0,0 +1,49 @@
+package seedu.address.commons.util;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Utilities about date and time
+ */
+public class DateTimeUtil {
+    /**
+     * Parse string into LocalDateTime
+     * @param str the string to be parsed
+     * @return The parsed LocalDateTime
+     * @throws DateTimeParseException If the string cannot be parsed into LocalDateTime
+     */
+    public static LocalDateTime parseString(String str) throws DateTimeParseException {
+        DateTimeFormatter formatter1 = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm[:ss]");
+        LocalDateTime result = null;
+        try {
+            result = LocalDateTime.parse(str);
+        } catch (DateTimeParseException e) {
+            try {
+                result = LocalDateTime.parse(str, formatter1);
+            } catch (DateTimeParseException e2) {
+                LocalDateTime now = LocalDateTime.now();
+                String appendDate = now.getYear() + "-" + now.getMonthValue() + "-" + now.getDayOfMonth() + " ";
+                try {
+                    result = LocalDateTime.parse(appendDate + str, formatter1);
+                } catch (DateTimeParseException e3) {
+                    String appendZero = "0";
+                    result = LocalDateTime.parse(appendDate + appendZero + str, formatter1);
+                }
```

Comment:
Seems too nested? Also, I am not sure what's happening in the last two try-catch clauses.